.github/workflows/ci-pipeline.yml

```
@@ -0,0 +1,22 @@
+name: Continuous Integration
+on:
+  push:
+    paths-ignore:
+      - "diagrams/*"
+      - "**/README.md"
+  pull_request:
+    paths-ignore:
+      - "diagrams/*"
+      - "**/README.md"
+
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
```

I know we're guilty of using/advertising this version, but we should actually be [using v4 now as per this](https://github.com/actions/checkout)!
```suggestion
        uses: actions/checkout@v4
```
(This will hopefully clear out some warnings on your pipeline)